frontend/src/context/auth-context.tsx

```diff
@@ -48,13 +60,50 @@ function AuthProvider({ children }: React.PropsWithChildren) {
   const clearGitHubToken = () => {
     setGitHubTokenState(null);
     localStorage.removeItem("ghToken");
+    localStorage.removeItem("ghTokenTime");
   };
 
   const logout = () => {
     clearGitHubToken();
     posthog.reset();
   };
 
+  const refreshToken = async () => {
+    const config = await OpenHands.getConfig();
+
+    if (config.APP_MODE !== "saas" || !gitHubTokenState) {
+      return;
+    }
+
+    let needsRefresh = false;
+
+    if (!gitHubTokenTime) {
+      // Refresh if the timestamp is missing
+      needsRefresh = true;
+    } else {
+      // Refresh if more than 7 hours old
+      const timeDiff = Date.now() - new Date(gitHubTokenTime).getTime();
+      needsRefresh = timeDiff > 7 * 60 * 60 * 1000;
```

Comment:
instead of tracking the time, we should attempt a refresh whenever we get a 401 or 403